metadata/src/test/java/org/apache/kafka/image/writer/ImageWriterOptionsTest.java

```diff
@@ -44,9 +48,33 @@ public void testSetMetadataVersion() {
                     setMetadataVersion(version);
             if (i < MetadataVersion.MINIMUM_BOOTSTRAP_VERSION.ordinal()) {
                 assertEquals(MetadataVersion.MINIMUM_KRAFT_VERSION, options.metadataVersion());
+                assertEquals(version, options.requestedMetadataVersion());
             } else {
                 assertEquals(version, options.metadataVersion());
             }
         }
     }
+
+    @Test
+    public void testHandleLoss() {
+        PrintStream originalOut = System.out;
+        String expectedMessage = "stuff";
+        Consumer<UnwritableMetadataException> customLossHandler = e -> System.out.println(e.getMessage());
+
+        for (int i = MetadataVersion.MINIMUM_KRAFT_VERSION.ordinal();
+             i < MetadataVersion.VERSIONS.length;
+             i++) {
+            ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+            MetadataVersion version = MetadataVersion.VERSIONS[i];
+            ImageWriterOptions options = new ImageWriterOptions.Builder()
+                    .setMetadataVersion(version)
+                    .setLossHandler(customLossHandler)
+                    .build();
+            System.setOut(new PrintStream(outContent));
+            options.handleLoss(expectedMessage);
+            System.setOut(originalOut);
+            String formattedMessage = String.format("Metadata has been lost because the following could not be represented in metadata version %s: %s", version, expectedMessage);
+            assertEquals(formattedMessage, outContent.toString().trim());
+        }
```

Comment:
I think we can simplify this test as this without relying System.out:
```
for (int i = MetadataVersion.MINIMUM_KRAFT_VERSION.ordinal();
             i < MetadataVersion.VERSIONS.length;
             i++) {
            
            MetadataVersion version = MetadataVersion.VERSIONS[i];
            String formattedMessage = String.format("Metadata has been lost because the following could not be represented in metadata version %s: %s", version, expectedMessage);
            Consumer<UnwritableMetadataException> customLossHandler = e -> {
                assertEquals(formattedMessage, e.getMessage());
            };
            ImageWriterOptions options = new ImageWriterOptions.Builder()
                    .setMetadataVersion(version)
                    .setLossHandler(customLossHandler)
                    .build();
            options.handleLoss(expectedMessage);
}
```

WDYT?

Reply:
@showuon Thanks! Yes it's a lot cleaner. I've adopted your suggestion =D